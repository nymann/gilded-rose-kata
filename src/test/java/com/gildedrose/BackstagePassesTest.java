package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.ClipboardReporter;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@UseReporter({ClipboardReporter.class})
public class BackstagePassesTest {

    @Test
    void qualityDropsToZeroAfterTheConcert() {
        Item item = new Item("Backstage passes to a TAFKAL80ETC concert", 0, 10);
        GildedRose gildedRose = new GildedRose(new Item[]{item});

        gildedRose.updateQuality();
        assertEquals(0, item.quality);
    }

    @Test
    void qualityIncreasesBy1WhenThereAreMoreThan10DaysRemaining() {
        Integer[] sellIn = {11};
        Integer[] qualities = {3};
        CombinationApprovals.verifyAllCombinations(this::updateQuality, sellIn, qualities);
    }

    @Test
    void qualityIncreasesBy2WhenThereAreMoreThan5ButLessThan10DaysRemaining() {
        Integer[] sellIns = {10, 9, 8, 7, 6};
        Integer[] qualities = {3, 48, 49};
        CombinationApprovals.verifyAllCombinations(this::updateQuality, sellIns, qualities);
    }

    @Test
    void qualityIncreasesBy3WhenThereAre5orLessDaysRemaining() {
        Integer[] sellIns = {5, 4, 3, 2, 1};
        Integer[] qualities = {10, 48};
        CombinationApprovals.verifyAllCombinations(this::updateQuality, sellIns, qualities);
    }

    private String updateQuality(int sellIn, int initialQuality) {
        Item item = new Item("Backstage passes to a TAFKAL80ETC concert", sellIn, initialQuality);
        GildedRose gildedRose = new GildedRose(new Item[]{item});

        gildedRose.updateQuality();

        return item.toString();
    }
}
