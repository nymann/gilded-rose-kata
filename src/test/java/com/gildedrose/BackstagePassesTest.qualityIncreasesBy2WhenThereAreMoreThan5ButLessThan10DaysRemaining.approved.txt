[10, 3] => Backstage passes to a TAFKAL80ETC concert, 9, 5 
[10, 48] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[9, 3] => Backstage passes to a TAFKAL80ETC concert, 8, 5 
[9, 48] => Backstage passes to a TAFKAL80ETC concert, 8, 50 
[9, 49] => Backstage passes to a TAFKAL80ETC concert, 8, 50 
[8, 3] => Backstage passes to a TAFKAL80ETC concert, 7, 5 
[8, 48] => Backstage passes to a TAFKAL80ETC concert, 7, 50 
[8, 49] => Backstage passes to a TAFKAL80ETC concert, 7, 50 
[7, 3] => Backstage passes to a TAFKAL80ETC concert, 6, 5 
[7, 48] => Backstage passes to a TAFKAL80ETC concert, 6, 50 
[7, 49] => Backstage passes to a TAFKAL80ETC concert, 6, 50 
[6, 3] => Backstage passes to a TAFKAL80ETC concert, 5, 5 
[6, 48] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
